trace-utils/src/send_data/mod.rs

```diff
@@ -63,6 +67,64 @@ pub struct SendData {
     target: Endpoint,
     headers: HashMap<&'static str, String>,
     retry_strategy: RetryStrategy,
+    #[cfg(feature = "compression")]
+    compression: Compression,
+}
+
+#[cfg(feature = "compression")]
+#[derive(Debug, Clone)]
+pub enum Compression {
+    Zstd(i32),
+    None,
+}
+
+pub struct SendDataBuilder {
+    pub(crate) tracer_payloads: TracerPayloadCollection,
+    pub(crate) size: usize,
+    target: Endpoint,
+    headers: HashMap<&'static str, String>,
+    retry_strategy: RetryStrategy,
+    #[cfg(feature = "compression")]
+    compression: Compression,
+}
+
+impl SendDataBuilder {
+    pub fn new(
+        size: usize,
+        tracer_payload: TracerPayloadCollection,
+        tracer_header_tags: TracerHeaderTags,
+        target: &Endpoint,
+    ) -> SendDataBuilder {
+        let mut headers: HashMap<&'static str, String> = tracer_header_tags.into();
+        headers.insert(DATADOG_SEND_REAL_HTTP_STATUS_STR, "1".to_string());
+        SendDataBuilder {
+            tracer_payloads: tracer_payload,
+            size,
+            target: target.clone(),
+            headers,
+            retry_strategy: RetryStrategy::default(),
+            #[cfg(feature = "compression")]
+            compression: Compression::None,
+        }
+    }
+
+    #[cfg(feature = "compression")]
+    pub fn with_compression(mut self, compression: Compression) -> SendDataBuilder {
+        self.compression = compression;
+        self
+    }
+
+    pub fn build(self) -> SendData {
```

Comment:
Forgive my rust-novice knowledge here, but consuming `self` here shouldn't change heap consumption, correct? We don't copy/clone any struct members so these are all references to the builder and should have the same scope when used.